### PR TITLE
better types

### PR DESCRIPTION
**Pull request by tbo47**
INSERT A SMALL DESCRIPTION OF WHAT YOU CHANGED HERE

#### **Verify that...**

- [x] `npm test` passes
- [x] `npm run api` passes
- [x] `npm run lint` passes
- [x] `npm run build` still works

### CODE REVIEW

File: src/og/Globe.js [hunk 1]
```diff
@@ -50,12 +50,12 @@ const PLANET_NAME_PREFIX = "globus_planet_";
  *
  * @param {object} options - Options:
  * @param {string} options.target - HTML element id where planet canvas have to be created.
- * @param {og.scene.RenderNode} [options.skybox] - Render skybox. null - default.
+ * @param {scene.RenderNode} [options.skybox] - Render skybox. null - default.
```
Comment (Zemledelec) [inline src/og/Globe.js]: Not scene.RenderNode but RenderNode

File: src/og/Globe.js [hunk 2]
```diff
@@ -50,12 +50,12 @@ const PLANET_NAME_PREFIX = "globus_planet_";
...
  * @param {object} options - Options:
  * @param {string} options.target - HTML element id where planet canvas have to be created.
- * @param {og.scene.RenderNode} [options.skybox] - Render skybox. null - default.
+ * @param {scene.RenderNode} [options.skybox] - Render skybox. null - default.
  * @param {string} [options.name] - Planet name. Default is unic identifier.
- * @param {og.terrain.Terrain} [options.terrain] - Terrain provider. Default no terrain - og.terrain.EmptyTerrain.
- * @param {Array.<og.control.Control>} [options.controls] - Renderer controls array.
- * @param {Array.<og.Layer>} [options.layers] - Planet layers.
- * @param {og.Extent} [options.viewExtent] - Viewable starting extent.
+ * @param {terrain.Terrain} [options.terrain] - Terrain provider. Default no terrain - og.terrain.EmptyTerrain.
```
Comment (Zemledelec) [inline src/og/Globe.js]: ...the same, just EmptyTerrain

File: src/og/Globe.js [hunk 3]
```diff
@@ -148,7 +148,7 @@ class Globe {
             /**
              * Render node renders a planet.
              * @public
-             * @type {og.scene.Planet|og.scene.PlanetAtmosphere}
+             * @type {scene.Planet|og.scene.PlanetAtmosphere}
```
Comment (Zemledelec) [inline src/og/Globe.js]: let's remove PlanetAtmosphere..it was long time ago:)

File: src/og/camera/Frustum.js [hunk 4]
```diff
@@ -227,7 +227,7 @@ class Frustum {
     /**
      * Returns true if the frustum contains a bonding sphere, but bottom plane exclude.
      * @public
-     * @param {og.bv.Sphere} sphere - Bounding sphere.
+     * @param {bv.Sphere} sphere - Bounding sphere.
```
Comment (Zemledelec) [inline src/og/camera/Frustum.js]: no bv just Sphere

**Review by Zemledelec — CHANGES_REQUESTED** [review]
Hi @tbo47 lets make type specification in module style as you do, but there is a problem I mentioned in a few comments. Check it out, please. Type specification needs a little more care in a few code files.

File: src/og/Globe.js (hunk 1 above)
Comment (tbo47) [reply src/og/Globe.js]: done

**Comment by tbo47** [comment]
> Hi @tbo47 lets make type specification in module style as you do, but there is a problem I mentioned in a few comments. Check it out, please. Type specification needs a little more care in a few code files.

it's now ready for review

File: src/og/Globe.js [hunk 5]
```diff
@@ -148,7 +152,7 @@ class Globe {
             /**
              * Render node renders a planet.
              * @public
-             * @type {og.scene.Planet|og.scene.PlanetAtmosphere}
+             * @type {Planet|og.scene.PlanetAtmosphere}
```
Comment (Zemledelec) [inline src/og/Globe.js:155]: Just for future we will remove og.scene.PlanetAtmosphere

File: src/og/camera/PlanetCamera.js [hunk 6]
```diff
@@ -18,19 +18,24 @@ import { Mat4 } from "../math/Mat4.js";
...
- * @param {og.Vec3} [options.look] - Camera look position. Default (0,0,0)
- * @param {og.Vec3} [options.up] - Camera eye position. Default (0,1,0)
+ * @param {Vec3} [options.eye] - Camera eye position. Default (0,0,0)
+ * @param {Vec3} [options.look] - Camera look position. Default (0,0,0)
+ * @param {Vec3} [options.up] - Camera eye position. Default (0,1,0)
  */
 class PlanetCamera extends Camera {
+
+    /**
+     * @param {RenderNode} planet - Planet render node.
```
Comment (Zemledelec) [inline src/og/camera/PlanetCamera.js:36]: It's doubled from 21. Looks strange. @tbo47, are you sure, it is correct?

File: src/og/control/CompassButton.js [hunk 7]
```diff
@@ -86,6 +86,11 @@ let svg = `<?xml version="1.0" encoding="UTF-8" standalone="no"?>
  * @params {Object} [options] - Control options.
  */
 class CompassButton extends Control {
+
+    /**
+     * 
+     * @params {Object} [options] - Control options.
```
Comment (Zemledelec) [inline src/og/control/CompassButton.js:92]: ...the same question. When class extends a class parameters are doubled. Is it ok?

File: src/og/ellipsoid/Ellipsoid.js [hunk 8]
```diff
@@ -15,6 +15,10 @@ import { LonLat } from '../LonLat.js';
  * @param {number} polarSize - Polar ellipsoid size.
  */
 class Ellipsoid {
+    /**
+     * @param {number} equatorialSize - Equatorial ellipsoid size.
```
Comment (Zemledelec) [inline src/og/ellipsoid/Ellipsoid.js:19]: May be we have to choose a place where we are going to make description obout constructor options. I prefer any place, but not two places.

File: src/og/utils/shared.js [hunk 9]
```diff
@@ -332,12 +332,12 @@ export function binaryInsert(ar, el, compare_fn) {
 /**
  * Returns two segment lines intersection coordinate.
  * @static
- * @param {og.math.Vec2} start1 - First line first coordinate.
- * @param {og.math.Vec2} end1 - First line second coordinate.
- * @param {og.math.Vec2} start2 - Second line first coordinate.
- * @param {og.math.Vec2} end2 - Second line second coordinate.
+ * @param {math.Vec2} start1 - First line first coordinate.
```
Comment (Zemledelec) [inline src/og/utils/shared.js:335]: When we will use this module lets remove math from Vec2. In future. Right now it is ok:)

File: src/og/scene/Planet.js [hunk 10]
```diff
@@ -212,50 +212,50 @@ export class Planet extends RenderNode {
...
+         * @type {quadTree.Node}
          */
         this._renderedNodes = [];
         this._renderedNodesInFrustum = [];
 
         /**
          * Created nodes cache
          * @protected
-         * @type {og.quadTree.Node}
+         * @type {quadTree.Node}
```
Comment (Zemledelec) [inline src/og/scene/Planet.js:223]: ...the same thing, just for future will remove quadTree, and idle etc.

**Review by Zemledelec — APPROVED** [review]
@tbo47 I've made a few comments. check it out for future. Looks good! Thanks!

**Pull request merged**